engine/src/main/java/io/zeebe/engine/processing/common/EventHandle.java

```diff
@@ -120,7 +120,7 @@ public void activateElement(
 
       } else {
         eventTriggerBehavior.activateTriggeredEvent(
-            catchEvent, eventScopeKey, elementRecord, variables);
+            catchEvent, elementRecord.getFlowScopeKey(), elementRecord, variables);
```

Comment:
Please help me understand. Is the eventScopeKey of a boundary event the service task? Is the elementRecord in this case the boundary event? And is the flow scope of the boundary event in this case the service task?

Reply:
> Is the eventScopeKey of a boundary event the service task?

Yes. In case of a boundary event, the parameter `eventScopeKey` was set to the key of the service task. But it should be the same flow scope as the service task.